src/main/kotlin/no/nav/syfo/behandler/BehandlerService.kt

```diff
@@ -165,14 +165,29 @@ class BehandlerService(
         behandler: Behandler,
         pBehandlerKontor: PBehandlerKontor,
     ) {
-        if (!behandler.kontor.system.isNullOrBlank() && pBehandlerKontor.system != behandler.kontor.system) {
-            updateBehandlerKontorSystem(behandler.kontor.partnerId, behandler.kontor.system)
+        if (shouldUpdateKontorSystem(behandler.kontor, pBehandlerKontor)) {
```

Comment:
Jeg har et elsk-hat forhold til ordet should, for det kan bety så mye. Men det er jo alltid en grad av kanskje/sannsynlighet involvert, så her fungerer det!